inspirehep/modules/records/api.py

```diff
@@ -600,22 +600,25 @@ def _get_ref(self):
         return absolute_url(u'/api/{endpoint}/{control_number}'.format(endpoint=endpoint,
                                                                        control_number=pid_value))
 
-    def _query_citing_records(self):
+    def _query_citing_records(self, session=None):
         """Returns records which cites this one."""
+        if not session:
+            session = db.session
         ref = self._get_ref()
         if not ref:
             raise Exception("There is no $ref for this object")
-        citations = RecordMetadata.query.with_entities(RecordMetadata.id).filter(
+        citations = session.query(RecordMetadata).with_entities(RecordMetadata.id).filter(
             referenced_records(RecordMetadata.json).contains([ref]))
         return citations
 
     @property
     def get_citing_records_query(self):
         return self._query_citing_records()
```

Comment:
we can remove this method

Reply:
Which one? Both `_query_citing_records` and `get_citing_records_query` are used in the code.

Reply:
the property ``get_citing_records_query`` we don't handle the session

Reply:
But we use it in the `LiteratureCitationsResource` view, and in there we do not need to handle the session as it always should be correct. Replacing it for the `_query_citing_records` in this view would also require to change `_query_citing_records` method name to `query_citing_records` which would also require to change call to it in `get_citations_count`. Do we really want to do that?

Reply:
I missed ``LiteratureCitationsResource`` 👍 ignore my comment